eng/scripts/Invoke-GenerateAndBuildV2.ps1

```diff
@@ -108,6 +109,39 @@ if ($inputFileToGen) {
     UpdateExistingSDKByInputFiles -inputFilePaths $inputFileToGen -sdkRootPath $sdkPath -headSha $commitid -repoHttpsUrl $repoHttpsUrl -downloadUrlPrefix "$downloadUrlPrefix" -generatedSDKPackages $generatedSDKPackages
 }
 
+# generate sdk from cadl file
+if ($relatedCadlProjectFolder) {
+    for ($i = 0; $i -le $relatedCadlProjectFolder.Count - 1; $i++) {
+        $caldFolder = (Join-Path $swaggerDir $relatedCadlProjectFolder[$i]) -replace "\\", "/"
```

Comment:
Why not just use a foreach. for statements in PowerShell are often not the best choice:

```powershell
foreach ($cadlFolder in $relatedCadlProjectFolder) {
  $cadlFolder = join-path $swaggerDir $cadlFolder)
```

Also, if you're having to replace backslashes with forward slashes to satisfy the tool, I would argue that's a bug in the tool. It should normalize and canonicalize folders itself as most tools do. Using it outside this script means it'll be a problem for people to type `{tools} foo{tab}` and use tab complete to complete directory/file names. That's a productivity issue.